Travis CI build is broken for most macOS versions

Example: https://travis-ci.org/beeware/rubicon-objc/builds/630497088

For all Travis environments except `xcode10.2` (the most recent one that our CI configuration includes),  the installation of Python 3.5 (using `pyenv`) fails with this message:

```
Last 10 log lines:
    import pip
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmprxyor7dg/pip-9.0.1-py2.py3-none-any.whl/pip/__init__.py", line 26, in <module>
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmprxyor7dg/pip-9.0.1-py2.py3-none-any.whl/pip/utils/__init__.py", line 27, in <module>
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmprxyor7dg/pip-9.0.1-py2.py3-none-any.whl/pip/_vendor/pkg_resources/__init__.py", line 35, in <module>
  File "/private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/python-build.20191231013638.23409/Python-3.5.6/Lib/plistlib.py", line 65, in <module>
    from xml.parsers.expat import ParserCreate
  File "/private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/python-build.20191231013638.23409/Python-3.5.6/Lib/xml/parsers/expat.py", line 4, in <module>
    from pyexpat import *
ImportError: No module named 'pyexpat'
make: *** [install] Error 1
```

I haven't looked into this in detail yet. Clearly the `pyexpat` module is missing, which likely means that it was not built during the CPython build. This usually happens when a required native library is not installed or cannot be found during build. I don't know what needs to be done to fix this - best case we only need to install the missing library with `brew`, but it might be more complicated than that (especially because all of these build environments were previously working).

I also don't know if this problem is specific to Python 3.5 or if it also affects later versions. Right now it's impossible to tell, because Python 3.5 is installed first, and the installation failure stops the entire build.

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,7 +24,7 @@ matrix:
     - osx_image: xcode7.3
     # macOS 10.13 High Sierra
     - osx_image: xcode10.1
-    # macOS 10.13 Mojave
+    # macOS 10.14 Mojave
     - osx_image: xcode10.2
     # macOS 10.12 Sierra
     - osx_image: xcode9.2

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,11 +50,9 @@ install:
   # pyenv's shims directory is not in the PATH by default.
   - export PATH="$(pyenv root)/shims:${PATH}"
   # These are the Python versions that we want to test on.
-  # The CFLAGS prefix is to set up the include path so that zlib is available to the Python build
-  # This is needed on Mojave, but doesn't hurt other versions.
-  - CFLAGS="-I$(xcrun --show-sdk-path)/usr/include" pyenv install --skip-existing ${PY35}
-  - CFLAGS="-I$(xcrun --show-sdk-path)/usr/include" pyenv install --skip-existing ${PY36}
-  - CFLAGS="-I$(xcrun --show-sdk-path)/usr/include" pyenv install --skip-existing ${PY37}
+  - pyenv install --skip-existing ${PY35}
+  - pyenv install --skip-existing ${PY36}
+  - pyenv install --skip-existing ${PY37}
   - pyenv global ${PY37} ${PY36} ${PY35} system
   - pip install --upgrade tox
   - make -f Makefile

